unit-tests/live/syncer/test-throughput.cpp

```diff
@@ -0,0 +1,248 @@
+
+//#cmake: static!
+//#test:device D400*
+//#test:device L500*
+
+
+#define CATCH_CONFIG_MAIN
+#include <stdlib.h> 
+#include "../../catch.h"
+#include "../../unit-tests-common.h"
+
+using namespace rs2;
+constexpr int RECEIVE_FRAMES_TIME = 10;
+
+TEST_CASE("Syncer dynamic FPS - throughput test", "[live]")
+{
+    typedef enum configuration
+    {
+        IR_ONLY,
+        IR_RGB_EXPOSURE,
+        STOP
+    }configuration;
+
+    class ir_cfg
+    {
+    public:
+        ir_cfg(sensor rgb_sensor,
+            sensor ir_sensor,
+            std::vector<rs2::stream_profile > rgb_stream_profile,
+            std::vector<rs2::stream_profile > ir_stream_profile) : _rgb_sensor(rgb_sensor), _ir_sensor(ir_sensor),
+            _rgb_stream_profile(rgb_stream_profile), _ir_stream_profile(ir_stream_profile) 
+        {
+            _delta = 0.0f;
+        }
+         
+        void test_configuration(configuration test)
+        {
+            std::cout << "==============================================" << std::endl;
+            std::string cfg = test == IR_ONLY ? "IR Only" : "IR + RGB";
+            std::cout << "Configuration: " << cfg << std::endl << std::endl;
+            
+            start_streaming(test);
+            process_validate_frames(test);
+            stop_streaming(test);
+            // analysis
+            analysis(test, _rgb_frames_arrival_info, "Color");
+            analysis(test, _ir_frames_arrival_info);
+        }
+
+    private:
+        void set_ir_exposure()
+        {
+            if (_ir_sensor.supports(RS2_OPTION_EXPOSURE))
+                REQUIRE_NOTHROW(_ir_sensor.set_option(RS2_OPTION_EXPOSURE, 18000)); // set exposure value to get fps = 30
+                //REQUIRE_NOTHROW(_ir_sensor.set_option(RS2_OPTION_EXPOSURE, 60000)); // set exposure value to x > 1000/fps to get fps = 15
+        }
+        void unset_ir_exposure()
+        {
+            if (_ir_sensor.supports(RS2_OPTION_EXPOSURE))
+                REQUIRE_NOTHROW(_ir_sensor.set_option(RS2_OPTION_EXPOSURE, 1)); // set exposure value to get fps = 60
+                //REQUIRE_NOTHROW(_ir_sensor.set_option(RS2_OPTION_EXPOSURE, 18000)); // set exposure value to get fps = 30
+        }
+        void start_streaming(configuration test)
+        {
+            _ir_sensor.set_option(RS2_OPTION_EXPOSURE, 1);
+            _ir_sensor.open(_ir_stream_profile); // ir streams in all configurations
+            _ir_sensor.start(_sync);
+            if (test == IR_RGB_EXPOSURE)
+            {
+                _rgb_sensor.open(_rgb_stream_profile);
+                _rgb_sensor.start(_sync);
+            }
+        }
+        void stop_streaming(configuration test)
+        {
+            _ir_sensor.stop();
+            _ir_sensor.close();
+            if (test == IR_RGB_EXPOSURE)
+            {
+                _rgb_sensor.stop();
+                _rgb_sensor.close();
+            }
+            unset_ir_exposure();
```

Comment:
Not sure if needed

Reply:
It is needed because in next configuration we need to start from low exposure then increase it. Without this line the next configuration starts with high exposure (18000)